src/extension-impl.ts

```diff
@@ -13,6 +13,7 @@ class ExtensionImpl extends Extension {
 		addProgramFlags(program, kaid || "");
 		addProgramDates(program, kaid || "");
 		if (kaid) {
```

Comment:
As of right now, your fork tells users who are not logged in that their vote was recorded when they press the upvote button.

`kaid` never evaluates as falsy in the current setup.  I'm quite sure that replacing `kaid` with `kaid && kaid !== program.kaid` here would fix the issue.

Reply:
In that case, you'll probably need to add a function to extension.ts that looks something like this:
```
const isLoggedIn = async (): Promise<boolean> => {
	const kaScript = await KAdefine.asyncRequire(KAScripts.KA);
	return !!(kaScript.isLoggedIn && kaScript.isLoggedIn());
};
```